### DISCUSSION TITLE
Scale down to 0 before database migration

### DISCUSSION THREAD

**Discussion by vrachok-sc** [fluxcd/flux2#2713] [Q&A] [answered]
Hello

Before migrating the database, I need to scale down my application replicas to zero, is it possible to do this using Flux?

Thank you in advance.

**Comment by stefanprodan** [comment, marked as answer]
Flux is for defining your cluster state in Git. If you commit a change with `replicas: 0` in Git, the Flux will apply it on the cluster and Kubernetes will scale that deployment to zero. You could build some kind of automation with GitHub Actions or some other CI, that does the YAML patching in Git (scale down, update migration job), waits for Flux to apply and run the migration, then it commits the scale up of the deployment.
